<?xml version="1.0"?>
<launch>
  <arg name="robot_name" default="tianracer" />
<<<<<<< HEAD
  <group ns="$(arg robot_name)">
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" clear_params="true">
      <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />

=======
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="laser_frame_id"   default="laser"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />
  <arg name="map_frame" default="odom" />
  <arg name="map_topic" default="odom" />
  <group ns="$(arg robot_name)">
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" clear_params="true">
      <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
      <param name="controller_frequency" value="10.0"/> 
>>>>>>> origin/feature_one_sim
      <rosparam file="$(find tianracer_gazebo)/config/teb/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find tianracer_gazebo)/config/teb/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find tianracer_gazebo)/config/teb/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find tianracer_gazebo)/config/teb/global_costmap_params.yaml" command="load" />

      <rosparam file="$(find tianracer_gazebo)/config/teb/move_base_params.yaml" command="load" />
      <rosparam file="$(find tianracer_gazebo)/config/teb/base_global_planner_params.yaml" command="load" />
      <rosparam file="$(find tianracer_gazebo)/config/teb/teb_local_planner_params.yaml" command="load" />
<<<<<<< HEAD
=======

      <!-- reset frame_id parameters using user input data -->
      <param unless="$(eval robot_name=='/')" name="local_costmap/robot_base_frame" value="$(arg robot_name)/$(arg base_frame_id)"/>
      <param unless="$(eval robot_name=='/')" name="local_costmap/global_frame" value="$(arg robot_name)/$(arg odom_frame_id)"/>
<!--      <param unless="$(eval robot_name=='/')" name="local_costmap/static_layer/map_topic" value="$(arg robot_name)/$(arg map_topic)"/>-->
      <param unless="$(eval robot_name=='/')" name="local_costmap/scan/sensor_frame" value="$(arg robot_name)/l$(arg laser_frame_id)"/>
      <param unless="$(eval robot_name=='/')" name="local_costmap/scan/topic" value="/$(arg robot_name)/$(arg laser_topic)"/>
      <param unless="$(eval robot_name=='/')" name="global_costmap/robot_base_frame" value="$(arg robot_name)/$(arg base_frame_id)"/>
      <param unless="$(eval robot_name=='/')" name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
<!--      <param unless="$(eval robot_name=='/')" name="global_costmap/static_layer/map_topic" value="$(arg robot_name)/$(arg map_topic)"/>-->
      <param unless="$(eval robot_name=='/')" name="global_costmap/scan/sensor_frame" value="$(arg robot_name)/$(arg laser_frame_id)"/>
      <param unless="$(eval robot_name=='/')" name="global_costmap/scan/topic" value="/$(arg robot_name)/$(arg laser_topic)"/>
      <param unless="$(eval robot_name=='/')" name="TebLocalPlannerROS/odom_topic" value="/$(arg robot_name)/$(arg odom_topic)"/>
      <param unless="$(eval robot_name=='/')" name="TebLocalPlannerROS/map_frame" value="$(arg robot_name)/$(arg map_frame)"/>
>>>>>>> origin/feature_one_sim
    </node>
  </group>
  
</launch> 
